### DISCUSSION TITLE
Could corfu be configured to show the popup when there is only one exact match?

### DISCUSSION THREAD

**Discussion by galeo** [minad/corfu#396] [Q&A]
For example, the current prefix is `a>b*>c`, and the sole completion item is also `a>b*>c`. I found that corfu's popup has been unable to display. I briefly looked at case 1) "Single exactly matching candidate and no further completion is possible" in `corfu--exhibit` and found that in this case only `corfu-quit` or `corfu--done` can be executed.

Could this behavior be adjusted and corfu can be configured to display the popup in this situation?

Thanks.

**Comment by minad** [comment]
Sure, we could add the value `show` to `corfu-on-exact-match`. We have to think about the exact semantics if auto completion or manual completion is used, and have to check if this makes sense. Let's try this. Feel free to propose a patch. Thanks!

**Comment by galeo** [comment]
> We have to think about the exact semantics if auto completion or manual completion is used, and have to check if this makes sense.

Here is a screenshot in vscode. As the emmet abbreviation is entered, the expanded markup is updated simultaneously. The entered prefix itself is the exact completion candidate.
<img width="643" alt="Screen Shot 2023-12-22 at 00 21 50" src="https://github.com/minad/corfu/assets/1025559/a71b520a-51b3-4f5c-9f14-a228d832cc00">

I just tried company-mode and it works. I think corfu should get this ability to support this scenario.

> Let's try this. Feel free to propose a patch.

I'll try it when I have time, but probably not too soon. I hope you can think about it. Thanks.

**Reply by minad** [reply]
> As the emmet abbreviation is entered, the expanded markup is updated simultaneously. The entered prefix itself is the exact completion candidate.

The Emmet auto expansion feature is interesting, but feels a bit like an abuse of the completion mechanism. I haven't tried the Emmet Emacs package for a long time. The last time I looked it wasn't maintained.

> I'll try it when I have time, but probably not too soon.

Looking forward to it.

**Comment by minad** [comment]
Resolved in #401
